src/types/composer.ts

```diff
@@ -1409,692 +1336,690 @@ export class TransactionComposer {
-      const suggestedParams = await this.getSuggestedParams()
-      // Build all of the transactions
-      const txnWithSigners: TransactionWithSignerAndContext[] = []
-      for (const txn of this.txns) {
-        txnWithSigners.push(...(await this.buildTxnWithSigner(txn, suggestedParams)))
+      return {
+        requiredFeeDelta,
+        unnamedResourcesAccessed: analysisParams.populateAppCallResources ? simulateTxnResult.unnamedResourcesAccessed : undefined,
       }
+    })
 
-      // Add all of the transactions to the underlying ATC
-      const methodCalls = new Map<number, algosdk.ABIMethod>()
-      txnWithSigners.forEach(({ context, ...ts }, idx) => {
-        this.atc.addTransaction(ts)
+    const sortedResources = groupResponse.unnamedResourcesAccessed
 
-        // Populate consolidated set of all ABI method calls
-        if (context.abiMethod) {
-          methodCalls.set(idx, context.abiMethod)
-        }
+    // NOTE: We explicitly want to avoid localeCompare as that can lead to different results in different environments
+    const compare = (a: string | bigint, b: string | bigint) => (a < b ? -1 : a > b ? 1 : 0)
 
-        if (context.maxFee !== undefined) {
-          this.txnMaxFees.set(idx, context.maxFee)
-        }
+    if (sortedResources) {
+      sortedResources.accounts?.sort((a, b) => compare(a.toString(), b.toString()))
+      sortedResources.assets?.sort(compare)
+      sortedResources.apps?.sort(compare)
+      sortedResources.boxes?.sort((a, b) => {
+        const aStr = `${a.app}-${a.name}`
+        const bStr = `${b.app}-${b.name}`
+        return compare(aStr, bStr)
+      })
+      sortedResources.appLocals?.sort((a, b) => {
+        const aStr = `${a.app}-${a.account}`
+        const bStr = `${b.app}-${b.account}`
+        return compare(aStr, bStr)
+      })
+      sortedResources.assetHoldings?.sort((a, b) => {
+        const aStr = `${a.asset}-${a.account}`
+        const bStr = `${b.asset}-${b.account}`
+        return compare(aStr, bStr)
       })
-      this.atc['methodCalls'] = methodCalls
     }
 
-    return { atc: this.atc, transactions: this.atc.buildGroup(), methodCalls: this.atc['methodCalls'] }
+    return {
+      transactions: txnAnalysisResults,
+      unnamedResourcesAccessed: analysisParams.populateAppCallResources ? sortedResources : undefined,
+    }
   }
 
   /**
    * Rebuild the group, discarding any previously built transactions.
    * This will potentially cause new signers and suggested params to be used if the callbacks return a new value compared to the first build.
-   * @returns The newly built atomic transaction composer and the transactions
+   * @returns The newly built transaction composer and the transactions
    * @example
    * ```typescript
    * const { atc, transactions, methodCalls } = await composer.rebuild()
    * ```
    */
   async rebuild() {
-    this.atc = new algosdk.AtomicTransactionComposer()
+    this.transactionsWithSigners = undefined
```

Comment:
Do we need to also do this.signedTransactions?

If so, i think it make sense to extract this logic into a private `reset()` method, which can be called by `send()` (in the scenario where resource population/fee coverage settings are different)  as well.